homework-submission/react-hyf/client/src/components/week3-homework/week3.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import User from './user/user';
+import Spinner from './spinner/spinner';
+import './week3.css';
+
+class Week3 extends Component {
+  state = { users: [], loading: true, show: true, error: '' };
+
+  componentDidMount() {
+    this.setFocus();
+    this.setState({ show: true });
+  }
+
+  handleKeyPress = e => {
+    if (e.key === ' ' || e.key === 'Spacebar') {
+      fetch(' https://uinames.com/api/?amount=10')
+        .then(res => res.json())
+        .then(result => {
+          const amount = result.slice(1, 2);
+          this.setState({ users: amount, loading: false, show: false });
+        })
+        .catch(err => {
+          console.log(err.message);
```

Comment:
you have to update the state here in case fetch throws an error, otherwise the user will always see a spinner in case of error too

Reply:
all issues have been fixed